### DISCUSSION TITLE
Meeting 2024-10-01

### DISCUSSION THREAD

**Discussion by jamesmunns** [rust-embedded/wg#797] [General]
This is the agenda for the next meeting. Please add anything you'd like to discuss below!

* Meetings: Tuesday 8pm Europe/Berlin time
* [Join the Matrix chat](https://matrix.to/#/#rust-embedded:matrix.org)
* [Nominated issues](https://github.com/search?q=org%3Arust-embedded+label%3Anominated+is%3Aopen&type=Issues)
* [IRC/Matrix logs](https://libera.irclog.whitequark.org/rust-embedded/2024-10-01)

### Announcements

* Next book sprint for the Discovery MB2 book will be on October 26, 17:00-21:00 CEST, 15:00-19:00 UTC, 8:00AM-12:00PM PST. Join if you're interested in helping to write the new discovery book! Give a thumbs up to [this comment](https://github.com/rust-embedded/wg/discussions/797#discussioncomment-10812164) to express your interest in joining.

## Carry over from last week:

There are three larger topics we should schedule some time for:

1. The discussion regarding the CS Mutex, and whether it needs an inner UnsafeCell. Context: https://github.com/rust-embedded/critical-section/issues/52 - we might need HAL team folks present.
2. We should follow up with discussing potential project goal proposals: https://github.com/rust-embedded/wg/issues/775 - relevant to whole WG
3. Discussion re: MMIO references. Context: https://github.com/rust-embedded/wg/issues/791 - we might also need the HAL team present

**Comment by jamesmunns** [comment]
CC @rust-embedded/hal - would y'all be able to be present/active in the meeting next week? There are two items above that are relevant and might be good to push forward.

**Reply by eldruin** [reply]
I will not be able to participate next week.

**Comment by jannic** [comment]
Another carry over from the week before: Is there anything blocking a cortex-m-rt release? https://github.com/rust-embedded/cortex-m/pull/533

**Comment by hdoordt** [comment]
Announcement: next book sprint for the Discovery MB2 book will be on October 26, 17:00-21:00 CEST, 15:00-19:00 UTC, 8:00AM-12:00PM PST. Join if you're interested in helping to write the new discovery book!

If you're interested in joining, you can make that clear by liking this message.

**Comment by BartMassey** [comment]
EuroRust will be happening on 10-08 and GOSIM Beijing will be happening on 10-15. I suspect a lot of us will be at one or the other. Shall we meet on Matrix those weeks? If so, who can run these meetings?

**Reply by jamesmunns** [reply]
I'll be available, and I think @adamgreig will be back as well. If anyone thinks we should skip I'm open, otherwise I think we'll have hands to manage it!

**Comment by rursprung** [comment]
**how to deal with long-term 0.x dependencies (defmt, heapless, etc.)?**

as per the [API guideline C-STABLE](https://rust-lang.github.io/api-guidelines/necessities.html#c-stable) a crate should only publish a stable release if all public dependencies are also stable. in the embedded rust ecosystem this is a bit hard since some core crates are not yet on 1.0 (e.g. https://github.com/knurling-rs/defmt/issues/795 & https://github.com/rust-embedded/heapless/issues/410).

does anyone know if these crates will soon(-ish) reach v1.0?
if not: are there general guidelines on how to deal with this? e.g. is it generally accepted if you e.g. do an 1.0 release as long as these 0.x dependencies are optional and you add a statement to this regard to the README (excluding them from usual semver guarantees, i.e. you can bump from 0.x to 0.y of an optional dependency in a minor release of your own crate)?

this comes from here, where i'm wondering whether i should release this as v0.3.0 (and wait indefinitely with 1.0 until all dependencies are on 1.0) or just go with v1.0 with a semver exclusion disclaimer: https://github.com/rust-embedded-community/adafruit-bluefruit-protocol-rs/pull/38 (in other news: any reviewers for this PR? 😉)
